litestar/handlers/http_handlers/base.py

```diff
@@ -499,12 +497,15 @@ def _validate_handler_function(self) -> None:
                 "If the function should return a value, change the route handler status code to an appropriate value.",
             )
 
+        if (
+            before_request := self.resolve_before_request()
+        ) and not before_request.parsed_signature.return_type.is_optional:
+            return_type = before_request.parsed_signature.return_type
+
         if not self.media_type:
-            if return_type.annotation is AnyStr or return_type.is_subclass_of(
-                (str, bytes, Redirect, File, FileResponse)
-            ):
+            if return_type.is_subclass_of((str, bytes)) or return_type.annotation is AnyStr:
                 self.media_type = MediaType.TEXT
-            else:
+            elif not return_type.is_subclass_of(Response):
```

Comment:
This change means that if a handler is annotated with a `Response` return type, then the handler media type is left empty. This lets me identify a media type that is explicitly set when merging response data with layered parameters for handlers that return `Response`, etc.